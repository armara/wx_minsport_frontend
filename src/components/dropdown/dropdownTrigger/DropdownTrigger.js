import React from 'react'
import PropTypes from 'prop-types'

import Button from 'components/button/Button'

import 'components/dropdown/dropdownTrigger/DropdownTrigger.scss'

const DropdownTrigger = ({ selectedItem, onClick }) => (
  <Button
    type="button"
    aria-haspopup="true"
    aria-controls="dropdown-menu2"
    onClick={onClick}
    className="dropdown-trigger"
  >
    <>
      <div>{selectedItem}</div>
      <span className="icon is-small">
        <i className="ic-keyboard-arrow-down" aria-hidden="true" />
      </span>
    </>
  </Button>
)

DropdownTrigger.propTypes = {
  selectedItem: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.number,
  ]).isRequired,
  onClick: PropTypes.func.isRequired,
}

export default DropdownTrigger
