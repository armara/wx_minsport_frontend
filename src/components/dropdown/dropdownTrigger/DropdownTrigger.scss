@import "styles/mixins";

.dropdown-trigger {
  display: flex;
  justify-content: space-between;
  align-items: center;
}