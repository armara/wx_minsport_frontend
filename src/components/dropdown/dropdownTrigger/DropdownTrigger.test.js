import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import DropdownTrigger from 'components/dropdown/dropdownTrigger/DropdownTrigger'

configure({ adapter: new Adapter() })

describe('<DropdownTrigger />', () => {
  it('renders correctly default props', () => {
    const tree = renderer
      .create(
        <DropdownTrigger onClick={() => {}} selectedItem="Peter the Great" />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
