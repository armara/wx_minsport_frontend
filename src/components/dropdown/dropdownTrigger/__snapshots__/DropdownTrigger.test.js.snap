// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownTrigger /> renders correctly default props 1`] = `
<button
  className="dropdown-trigger button base-button"
  disabled={false}
  id={null}
  onClick={[Function]}
  title=""
  type="button"
>
  
  <div>
    Peter the Great
  </div>
  <span
    className="icon is-small"
  >
    <i
      aria-hidden="true"
      className="ic-keyboard-arrow-down"
    />
  </span>
</button>
`;
